serializers/src/main/java/io/pravega/schemaregistry/serializers/EncodingCache.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.serializers;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.pravega.common.Exceptions;
+import io.pravega.schemaregistry.client.SchemaRegistryClient;
+import io.pravega.schemaregistry.client.exceptions.RegistryExceptions;
+import io.pravega.schemaregistry.contract.data.EncodingId;
+import io.pravega.schemaregistry.contract.data.EncodingInfo;
+import lombok.Data;
+
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Local cache for storing schemas that are retrieved from the registry service.  
+ */
+public class EncodingCache {
+    private static final Duration EXPIRY_AFTER_ACCESS = Duration.ofMinutes(20);
```

Comment:
If there is no change to the encoding, then even if the client is idle, there shouldn't be any change to the cache content, is it right? If so, then isn't it the case that we should only drop it upon a replacement (new encoding info)?

Reply:
we could have number of elements based cache. 
the intent is to ensure that if there are too many schemas then we dont keep older schemas cached.